tests/test_backend.py

```diff
@@ -3108,6 +3108,63 @@ def func(x, y):
             return tf.cast(s_, tf.float32, name=_TFOUTPUT)
         self._run_test_case(func, [_OUTPUT], {_INPUT: x_val, _INPUT1: y_val})
 
+    @check_opset_min_version(11)
+    @check_tf_min_version("2.2")
+    def test_matrix_diag_part_v3(self):
+        x_val = np.random.random([2, 3, 4, 5]).astype(np.float32)
+
+        def func_v2(X, K):
+            return tf.raw_ops.MatrixDiagPartV2(input=X, k=K, padding_value=0.123, name=_TFOUTPUT)
+
+        def func_v3(X, K):
+            return tf.raw_ops.MatrixDiagPartV3(input=X, k=K, padding_value=0.123, align='RIGHT_LEFT', name=_TFOUTPUT)
+
+        k_val = np.array([0]).astype(np.int32)
+        self._run_test_case(func_v2, [_OUTPUT], {_INPUT: x_val, _INPUT1: k_val})
+        self._run_test_case(func_v3, [_OUTPUT], {_INPUT: x_val, _INPUT1: k_val})
```

Comment:
Maybe  a function like run_test_v2_v3(...) can reduce the number of lines here?

Reply:
Good point.

Reply:
lines trimmed by latest commit.